src/NServiceBus.Core.Tests/DelayedDelivery/RouteDeferredMessageToTimeoutManagerBehaviorTests.cs

```diff
@@ -18,12 +18,17 @@ class RouteDeferredMessageToTimeoutManagerBehaviorTests
         {
             var behavior = new RouteDeferredMessageToTimeoutManagerBehavior("tm");
             var delay = TimeSpan.FromDays(1);
-            var message = new OutgoingMessage("id", new Dictionary<string, string>(), new byte[0]);
 
             var headers = new Dictionary<string, string>();
             string destination = null;
-            var context = new RoutingContext(message, new UnicastRoutingStrategy("target"), null);
-            context.AddDeliveryConstraint(new DelayDeliveryWith(delay));
+            var context = new TestableRoutingContext
+            {
+                RoutingStrategies = new List<RoutingStrategy>
```

Comment:
Isn't this a sign that https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Pipeline/Outgoing/IRoutingContext.cs#L20 shouldn't be a readonly collection?

If it's readonly why is there a setter?